src/test/ui/generic-associated-types/construct_with_other_type.stderr

```diff
@@ -2,11 +2,16 @@ error[E0271]: type mismatch resolving `for<'a> <<T as Baz>::Baa<'a> as std::ops:
   --> $DIR/construct_with_other_type.rs:19:9
    |
 LL | impl<T> Baz for T where T: Foo {
-   |         ^^^ expected type parameter `T`, found associated type
+   |      -  ^^^ expected type parameter `T`, found associated type
+   |      |
+   |      this type parameter
    |
    = note: expected associated type `<T as Foo>::Bar<'_, 'static>`
               found associated type `<<T as Baz>::Quux<'_> as Foo>::Bar<'_, 'static>`
-   = note: you might be missing a type parameter or trait bound
+help: consider further restricting this bound
+   |
+LL | impl<T> Baz for T where T: Foo + Baz<Quux = T> {
+   |                                ^^^^^^^^^^^^^^^
```

Comment:
This suggestion will  cause another error:

```
error[E0275]: overflow evaluating the requirement `<T as Baz>::Quux`
  --> src/test/ui/generic-associated-types/construct_with_other_type.rs:19:9
   |
19 | impl<T> Baz for T where T: Foo + Baz<Quux = T> {
   |         ^^^
   |
   = note: required because of the requirements on the impl of `Baz` for `T`
   = note: required because of the requirements on the impl of `Baz` for `T`
```

I feel this is "fine" since the output was _already_ confusing and the code itself is a bit contrived and requires `#![feature(generic_associated_types)]` to be enabled to even be triggered.